protocol/consumer/consumer.go

```diff
@@ -1,21 +1,85 @@
 package consumer
 
-const MaxVersionSupported = 1
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/segmentio/kafka-go/protocol"
+)
+
+const MaxVersionSupported = 3
 
 type Subscription struct {
-	Version         int16            `kafka:"min=v0,max=v1"`
-	Topics          []string         `kafka:"min=v0,max=v1"`
-	UserData        []byte           `kafka:"min=v0,max=v1,nullable"`
-	OwnedPartitions []TopicPartition `kafka:"min=v1,max=v1"`
+	Version         int16            `kafka:"min=v0,max=v3"`
+	Topics          []string         `kafka:"min=v0,max=v3"`
+	UserData        []byte           `kafka:"min=v0,max=v3,nullable"`
+	OwnedPartitions []TopicPartition `kafka:"min=v1,max=v3"`
+	GenerationID    int32            `kafka:"min=v2,max=v3"`
+	RackID          string           `kafka:"min=v3,max=v3,nullable"`
+}
+
+type subscriptionBackwardsCompat struct {
+	Version  int16    `kafka:"min=v0,max=v1"`
```

Comment:
Style wise I'm on the fence of adding an addition type for this vs. handling this in Subscriptions's FromBytes method. What do you think?